### PR TITLE
Hos

### PR DESCRIPTION
**Pull request by jstarck**
Creation of a class for multi-scale peaks count and wavelet-l1 norm statistics, taking into account the mask. 
Fix the bug on mass mapping for non square images.
Write two test routines to show to use the class 
   - test_hos_cfis()
   - test_hos_test1()

### CODE REVIEW

**Comment by AndreasTersenov** [comment]
There are a couple of small issues:

1. In `hos_peaks_l1.py` there is a semicolon (;) at the beginning of line 659 that should be removed (possibly this is what causes the check fail).
2. There are some instances in the repo where `np.float`  is used (I found some in `starlet.py` but there may be some in other places as well). This is deprecated since NumPy 1.20 so those instances should be replaced with either just `float` or `np.float64`. Doing this will not modify any behavior and is safe.
3.  In the test routine functions, in `hos_peaks_l1.py`, some directories paths are defined, but they use your local paths. If we want these test routines to be able to be ran by anyone who installs the package, this should be changed.

File: pycs/misc/stats.py [hunk 1]
```diff
@@ -403,6 +407,57 @@ def get_grf(size, intput_power_map=None):
             map_fourier[i, j] = power_sample[i, j]  #
             map_fourier[size - i, size - j] = np.conj(power_sample[i, j])
 
-    map_pixel = np.fft.ifft2(np.fft.fftshift(map_fourier))
+    map_pixel = np.fft.ifft2(np.fft.fftshift(map_fourier, axes=(-2, -1)))
 
     return map_pixel
+
+
+def mad(input_data, axis=None, keepdims=False):
```
Comment (sfarrens) [inline pycs/misc/stats.py:415]: The MAD is already implemented in ModOpt

File: pycs/astro/wl/mass_mapping.py [hunk 2]
```diff
@@ -108,25 +105,126 @@ def get_shear_noise(self, FillMask=False):
...
+#
+#         # Compute residual
+#         residual = (image - result)*mask
+#         # print("Iter ", ii+1, ": Err = ", np.std(residual))
+#         # Take a step
+#         result += residual
+#
+#     return result
+#
+# =============================================================================
```
Comment (sfarrens) [inline pycs/astro/wl/mass_mapping.py:227]: Why is the commented out. Code should either be in or out not half way between.

File: pycs/astro/wl/mass_mapping.py [hunk 3]
```diff
@@ -982,98 +1085,146 @@ def prox_wiener_filtering(
...
+        #         if Inpaint:
+        #             u1 = xg.real
+        #             u2 = xg.imag
+        #             u1 = new_dct_inpainting(u1, mask, niter=100, thresholding="hard", MultiScaleConstraint=None)
+        #             u2 = new_dct_inpainting(u2, mask, niter=100, thresholding="hard", MultiScaleConstraint=None)
+        #             xg = u1 + 1j * u2
+        #
+        #             tvilut(u1)
+        #             tvilut(u2)
+        # =============================================================================
```
Comment (sfarrens) [inline pycs/astro/wl/mass_mapping.py:1203]: Who is this for?

File: pycs/astro/wl/hos_peaks_l1.py [hunk 4]
```diff
@@ -0,0 +1,922 @@
...
+#         vmin (float): Minimum value for colormap scaling.
+#         vmax (float): Maximum value for colormap scaling.
+#     """
+#     plt.figure()
+#     img = plt.imshow(map_data, cmap=lut, vmin=vmin, vmax=vmax, origin='lower')
+#     plt.title(title)
+#     plt.colorbar(img)
+#     if filename is not None:
+#         plt.savefig(filename)
+#     plt.show()
```
Comment (sfarrens) [inline pycs/astro/wl/hos_peaks_l1.py:614]: In or out?

File: pycs/astro/wl/hos_peaks_l1.py [hunk 5]
```diff
@@ -0,0 +1,922 @@
...
+    H.plot_l1norm()
+
+
+############ END CLASS #######################
+
+# if __name__ == '__main__':
+#     print ( "Main :)")
+
+# test_hos_cfis()
+# test_hos_test1()
```
Comment (sfarrens) [inline pycs/astro/wl/hos_peaks_l1.py:922]: Remove

**Review by sfarrens — CHANGES_REQUESTED** [review]

**Pull request merged**